peps/pep-9999.rst

```diff
@@ -0,0 +1,922 @@
+primarily been served by the :mod:`ssl` module, which provides a Python API to the
+`OpenSSL library <https://www.openssl.org/>`_.
+
+Because the :mod:`ssl` module is distributed with the Python standard library,
+it has become the overwhelmingly most-popular method for handling TLS in Python.
+An extraordinary majority of Python libraries, both in the standard library and
+on the Python Package Index, rely on the ssl module for their TLS connectivity.
+
+Unfortunately, the preeminence of the :mod:`ssl` module has had a number of
+unforeseen side-effects that have had the effect of tying the entire Python
+ecosystem tightly to OpenSSL. This has forced Python users to use OpenSSL even
+in situations where it may provide a worse user experience than alternative TLS
+implementations, which imposes a cognitive burden and makes it hard to provide
+“platform-native” experiences.
+
+Problems
+--------
+
+The fact that the :mod:`ssl` module is built into the standard library has meant
+that all standard-library Python networking libraries are entirely reliant on
+the OpenSSL that the Python implementation has been linked against. This leads
+to the following issues:
+
+* It is difficult to take advantage of new, higher-security TLS without
+  recompiling Python to get a new OpenSSL. While there are third-party bindings
+  to OpenSSL (e.g. `pyOpenSSL <https://pypi.org/project/pyOpenSSL/>`_), these
+  need to be shimmed into a format that the standard library understands,
+  forcing projects that want to use them to maintain substantial compatibility
+  layers.
+
+* For Windows distributions of Python, they need to be shipped with a copy of
+  OpenSSL. This puts the CPython development team in the position of being
+  OpenSSL redistributors, potentially needing to ship security updates to the
+  Windows Python distributions when OpenSSL vulnerabilities are released.
+
+* For macOS distributions of Python, they need either to be shipped with a copy
+  of OpenSSL or linked against the system OpenSSL library. Apple has formally
+  deprecated linking against the system OpenSSL library, and even if they had
+  not, that library version has been unsupported by upstream for nearly one year
+  as of the time of writing. The CPython development team has started shipping
+  newer OpenSSLs with the Python available from python.org, but this has the
+  same problem as with Windows.
+
+* Users may wish to integrate with TLS libraries other than OpenSSL for other
+  reasons, such as maintenance burden versus a system-provided implementation,
+  or because OpenSSL is simply too large and unwieldy for their platform (e.g.
+  for embedded Python). Those users are left with the requirement to use
+  third-party networking libraries that can interact with their preferred TLS
+  library or to shim their preferred library into the OpenSSL-specific
+  :mod:`ssl` module API.
+
+Additionally, the ssl module as implemented today limits the ability of CPython
```

Comment:
```suggestion
Additionally, the :mod:`ssl` module as implemented today limits the ability of CPython
```
Or:
```suggestion
Additionally, the :mod:`!ssl` module as implemented today limits the ability of CPython
```
Since it's being linked to a lot; I would maybe only keep the first link in each section.